example_scenes/advanced_tex_fonts.py

```diff
@@ -0,0 +1,691 @@
+from manim.mobject.svg.tex_mobject import TextMobject
+from manim.utils.config_ops import digest_config
+
+# for the demo Scene
+from manim import (
+    BLACK,
+    WHITE,
+    UP,
+    UL,
+    ORIGIN,
+    DOWN,
+    LEFT,
+    RIGHT,
+    TextMobject,
+    ApplyMethod,
+    AnimationGroup,
+    VGroup,
+    FadeIn,
+    FadeOut,
+    Transform,
+    FadeOutAndShift,
+    FadeInFromDown,
+    FadeOutAndShiftDown,
+    Scene,
+)
+
+# Some font configurations specify the compiler as LaTeX, XeLaTeX, LuaLaTeX etc.
+# and also specify whether the code will compile to dvi or pdf format.
+# The goal is to read the font configuration from the TEX_FONT_CONFIGS dictionary
+# below and prepare a corresponding tex configration and template.
+
+# We then define FontMobject for convenience to easily add text to the demo scene
+# For example FontMobject("example", font="ecfaugieeg")
+
+# Then we define a demo scene that adds one line of every font.
+
+# WARNING: The demo scene will not compile unless all relevant fonts are
+# installed on your system. If a font (or LaTeX package) is missing, you will
+# get a Tex compilation error. You can then try to remebdy the error by installing
+# the missing font, or you can comment out the relevant entry in the TEX_FONT_CONFIGS
+# dictionary and try again without that font.
+
+
+class TexTemplateFromFontConfig(TexTemplateFromFile):
+    def __init__(self, font, **kwargs):
+        digest_config(self, kwargs)
+        self.font = font
+        if "compiler" in TEX_FONT_CONFIGS[font]:
+            tex_command = TEX_FONT_CONFIGS[font]["compiler"]
+        else:
+            tex_command = "latex"
+        if "output" in TEX_FONT_CONFIGS[font]:
+            tex_output = TEX_FONT_CONFIGS[font]["output"]
+        else:
+            tex_output = ".dvi"
+        self.tex_compiler = {
+            "command": tex_command,
+            "output_format": tex_output,
+        }
+        self.rebuild_cache()
+
+    def rebuild_cache(self):
+        self.body = self.GenerateTexTemplateForFont(self.font)
+
+    def GenerateTexTemplateForFont(self, font_name):
+        # TODO figure out how to do this with TexTemplate()
+        template = r"""
+\documentclass[preview]{standalone}
+
+\usepackage[english]{babel}
+\usepackage{amsmath}
+\usepackage{amssymb}
+
+"""
+        template += TEX_FONT_CONFIGS[font_name]["config"]
+        template += (
+            ""
+            if r"\begin{document}" in TEX_FONT_CONFIGS[font_name]["config"]
+            else r"""
+\begin{document}
+            """
+        )
+        template += r"""
+YourTextHere
+
+\end{document}
+"""
+        return template
+
+
+class FontMobject(TextMobject):
+    def __init__(self, *tex_strings, font=None, **kwargs):
+        if font is None:  # default to TextMobject
+            super().__init__(*tex_strings, **kwargs)
+        else:
+            super().__init__(
+                *tex_strings, template=TexTemplateFromFontConfig(font), **kwargs
+            )
```

Comment:
These classes don't have a `construct` function. This creates an issue as users usually run this file without scene names and manim consider them as Scenes and the user gets disappointed as manim will throw an error. AFAIK.
Possibly moving to somewhere else or making it hidden would do.

Reply:
Yeah, the example_scenes folder is meant to contain Scenes that the user can run out of the box.

Reply:
I didn't want to include it as-is in manim proper so I thought it can serve as an example file and people can take it from there if interested.


"or making it hidden would do."   -- what does this mean?

Reply:
Well, the intended purpose is to contain `Scene` classes, that's why it's called `example_scenes` folder. If this is more of a "this is the intended use", without actually rendering a scene, then it might find a better home as a tutorial in the documentaton.

Reply:
> "or making it hidden would do." -- what does this mean?

I thought `__all__ does it but IDK exactly.

Reply:
Ah. I took it as an example scene + context.
What I mean is, when you write your own scenes you will also write custom animations and text classes and so on, and here is a full example: it sets up tex template classes, set's up the FontMobject helper class and then renders an example scene.

but ok, if example_scenes is not the right place for this, where should we put it?

Reply:
A new page in docs.

Reply:
For the record:  `$ manim ./advanced_tex_fonts.py` works for me without issue